File: 프로그래머스/이상한문자만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string s) {
	string answer = "";

	int index = 0;

	for (int i = 0; i < s.size(); i++) {
		if (s[i] == ' ') {
			answer.push_back(' ');
			index = 0;
			continue;
		}
		if (index % 2 == 0) {
			if (s[i] > 90)
				answer.push_back(s[i] - 32);
			else
				answer.push_back(s[i]);
		}
		else {
			if (s[i] < 97)
				answer.push_back(s[i] + 32);
			else
				answer.push_back(s[i]);
		}
		index++;
	}

	return answer;
}

int main() {
	cout << solution("try hello world");
}

File: 프로그래머스/이상한문자만들기.py
def solution(s):
    clist = []
    for char in s.split(' '):
        idx = 0
        for i in char:
            if idx % 2 == 0:
                clist.append(i.upper())
            else:
                clist.append(i.lower())
            idx += 1
        clist.append(" ")
    clist.pop()

    return "".join(clist)


print(solution("try hello world"))
